### PR TITLE
Adapt translation to early return question mark

### PR DESCRIPTION
**Pull request by jschneider-bensch**
This PR addresses #562.

~It depends on changes in #550 and #559 so it is a draft until those are in.~

### CODE REVIEW

File: engine/backends/proverif/proverif_backend.ml [hunk 1]
```diff
@@ -278,6 +311,27 @@ module Make (Options : OPTS) : MAKE = struct
...
+                  (*   | TApp { ident = `TupleType _ } *)
+                  (*   (\* Tuple types should be translated without conversion from bitstring *\) *)
+                  (*     -> *)
+                  (*       iblock parens inner_field_doc *)
+                  (*   | _ -> *)
+                  (*       iblock parens *)
+                  (*         (inner_field_type_doc ^^ string "_to_bitstring" *)
+                  (*         ^^ iblock parens inner_field_doc) *)
+                  (* in *)
+                  (* string "Some" ^^ inner_block *)
```
Comment (W95Psp) [inline engine/backends/proverif/proverif_backend.ml:334]: Should we just remove this?

File: engine/backends/proverif/proverif_backend.ml [hunk 2]
```diff
@@ -278,6 +311,27 @@ module Make (Options : OPTS) : MAKE = struct
                           ^^ iblock parens inner_field_doc)
                   in
                   string "Some" ^^ inner_block
+              | PConstruct { name; args }
+              (* Some(expr) -> Some(<expr_typ_to_bitstring>(expr))*)
```
Comment (W95Psp) [inline engine/backends/proverif/proverif_backend.ml:315]: Can you rephrase the comment `handles "Some(expr) -> Some..."` or something? I actually thought this was commented code :sweat_smile:

**Review by W95Psp — APPROVED** [review]
A few nits, but otherwise that looks good to me :)

File: engine/backends/proverif/proverif_backend.ml (hunk 2 above)
Comment (jschneider-bensch) [reply engine/backends/proverif/proverif_backend.ml]: Oops, totally! I wrote a more sensible comment explaining what happens and why.
